### PR TITLE
Run markdown link checker only on markdown changes

### PR DESCRIPTION
**Pull request by sverchdotgov**
This makes sure that this action only runs if the pull request is on the main branch and is changing markdown files.

Fixes: https://github.com/usds/justice40-tool/issues/1114

### CODE REVIEW

File: .github/workflows/markdown-link-check.yml
Comment (lucasmbrown-usds) [inline .github/workflows/markdown-link-check.yml:4]: What was your motivation for restricting to `main`? I would think we would want to run this checker *before* merging to main.

Comment (sverchdotgov) [reply .github/workflows/markdown-link-check.yml:4]: I copied this part from the codeql checks, which seem to run on every push to a PR against main (see below check).

I suppose this one doesn't matter because main is protected, but it does mean that we won't have github actions run on people's dev branches if they aren't part of a PR yet.

Comment (esfoobar-usds) [reply .github/workflows/markdown-link-check.yml:4]: Push to `main` doesn't make sense in our repos since pushing to `main` is blocked. You should just use the `pull_request` action.

**Comment by sverchdotgov** [comment]
@esfoobar-usds Removed the "on.push" block.

File: .github/workflows/markdown-link-check.yml
Comment (lucasmbrown-usds) [inline .github/workflows/markdown-link-check.yml:5]: I still confused how we landed here. I thought I suggested that we want to check these *before* merging to main (so we don't introduce errors on main), and @esfoobar-usds suggested a very useful check that runs on pull requests? Did I miss some of the conversation? 

Thanks!

Comment (sverchdotgov) [reply .github/workflows/markdown-link-check.yml:5]: @lucasmbrown-usds I believe the configuration as it is now will run whenever someone creates a pull request that is intended to merge into main, so it will catch that case.

This just means if we create a PR against someone else's branch, these won't run.

See https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#example-using-multiple-events-with-activity-types-or-configuration

**Review by esfoobar-usds — APPROVED** [review]
🍏 approving for now, we'll review if it keeps getting errors

**Pull request merged**
